src/Microsoft.ML.CpuMath/CpuMathUtils.DotNetCoreApp.cs

```diff
@@ -0,0 +1,362 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Intrinsics.X86;
+using System;
+
+namespace Microsoft.ML.Runtime.Internal.CpuMath
+{
+    public static partial class CpuMathUtils
+    {
+        public const int CbAlign = 16;
+
+        private static bool Compat(AlignedArray a)
+        {
+            Contracts.AssertValue(a);
+            Contracts.Assert(a.Size > 0);
+            return a.CbAlign == CbAlign;
+        }
+
+        private unsafe static float* Ptr(AlignedArray a, float* p)
+        {
+            Contracts.AssertValue(a);
+            float* q = p + a.GetBase((long)p);
+            Contracts.Assert(((long)q & (CbAlign - 1)) == 0);
+            return q;
+        }
+
+        public static void Scale(float a, float[] dst, int count)
```

Comment:
I agree they are unclear.

But I don't think we should we should do a one-off change here and rename these variables, but the existing, corresponding variables in Sse.cs be left as `a`, `dst`.  If we intend on renaming, I think we should do a mass renaming across the board for all these functions.

Reply:
Right, I think this falls under my general comment of: A clean port at first is desirable, so a number of my comments can be read as "we should do this later in a follow up"

Reply:
Yes, I agree that they are unclear too.  Originally I named the first argument "scale" instead of "a", but the existing code used "a" so I followed the existing convention.  Thanks for raising the naming issue.

Reply:
> If we intend on renaming, I think we should do a mass renaming across the board for all these functions.

Yes. This should probably be done as a cleanup item.